packages/parser-babel-config/src/babel-config.ts

```diff
@@ -0,0 +1,110 @@
+import * as path from 'path';
+import * as ajv from 'ajv';
+
+import { ScanEnd, FetchEnd, Parser } from 'sonarwhal/dist/src/lib/types';
+import { Sonarwhal } from 'sonarwhal';
+import { loadJSONFile } from 'sonarwhal/dist/src/lib/utils/misc';
+
+import { BabelConfig, BabelConfigInvalid, BabelConfigParsed, BabelConfigInvalidSchema } from './types';
+
+export default class BabelConfigParser extends Parser {
+    private configFound: boolean = false;
+    private schema: any;
+    private validator: ajv.Ajv;
+
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+        this.schema = loadJSONFile(path.join(__dirname, 'schema', 'babelConfigSchema.json'));
+
+        /*
+         * If we want to use the ajv types in TypeScript, we need to import
+         * ajv in a lowsercase variable 'ajv', otherwise, we can't use types
+         * like `ajv.Ajv'.
+         */
+        this.validator = new ajv({ // eslint-disable-line new-cap
+            $data: true,
+            allErrors: true,
+            schemaId: 'id',
+            verbose: true
+        });
+
+        this.validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+
+        /**
+         * package.json => type: 'json' (file type from extention).
+         */
+        sonarwhal.on('fetch::end::json', this.parseBabelConfig.bind(this));
+        sonarwhal.on('scan::end', this.parseEnd.bind(this));
+    }
+
+    private async parseEnd(scanEnd: ScanEnd) {
+        if (!this.configFound) {
+            await this.sonarwhal.emitAsync('notfound::babel-config', scanEnd);
+        }
+    }
+
+    private async validateSchema(config: BabelConfig, resource: string) {
+        // ajv is lower case to be able to get the types.
+        const validate: ajv.ValidateFunction = this.validator.compile(this.schema);
+        const valid = validate(config);
+
+        if (!valid) {
+            const event: BabelConfigInvalidSchema = {
+                errors: validate.errors,
+                resource
+            };
+
+            await this.sonarwhal.emitAsync('invalid-schema::babel-config', event);
+        }
+
+        return valid;
+    }
+
+    private async parseBabelConfig(fetchEnd: FetchEnd) {
+        const resource = fetchEnd.resource;
+        const resourceFileName = path.basename(resource);
+        const isPackageJson: boolean = resourceFileName === 'package.json';
+        const isBabelrc: boolean = resourceFileName === '.babelrc';
+
+        if (!isBabelrc && !isPackageJson) {
+            return;
+        }
+
+        this.configFound = true; // It could be a `package.json` w/o `babel`, which will be verified below.
```

Comment:
Wait until you confirm that you have a babel configuration before set this variable to true.